### PR TITLE
feat: support `createNew` in `Deno.writeFile`

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: cli/tests/unit/write_file_test.ts [hunk 1]
```diff
@@ -96,6 +96,20 @@ Deno.test(
   },
 );
 
+Deno.test(
+  { permissions: { read: true, write: true } },
+  function writeFileSyncCreateNew() {
+    const enc = new TextEncoder();
+    const data = enc.encode("Hello");
+    const filename = Deno.makeTempDirSync() + "/test.txt";
```
Comment (dsherret) [inline cli/tests/unit/write_file_test.ts:104]: We should clean up this temp dir after running the test.

File: cli/tests/unit/write_file_test.ts [hunk 2]
```diff
@@ -216,6 +230,19 @@ Deno.test(
   },
 );
 
+Deno.test(
+  { permissions: { read: true, write: true } },
+  async function writeFileCreateNew() {
+    const enc = new TextEncoder();
+    const data = enc.encode("Hello");
+    const filename = Deno.makeTempDirSync() + "/test.txt";
```
Comment (dsherret) [inline cli/tests/unit/write_file_test.ts:238]: Same here.

**Review by dsherret — APPROVED** [review]
LGTM, but we should clean up the temp dir after running the tests.

File: cli/tests/unit/write_file_test.ts (hunk 1 above)
Comment (crowlKats) [reply cli/tests/unit/write_file_test.ts:104]: We dont do any cleaning up whatsoever with any of the other tests in this file; i would say to add some sort of cleanup it should be a follow-up

Comment (dsherret) [reply cli/tests/unit/write_file_test.ts:104]: Sure, a follow up sounds good. We should be cleaning up when running tests.

**Pull request merged**